snuba/subscriptions/executor_consumer.py

```diff
@@ -210,18 +220,38 @@ def __init__(
         entity_names: Sequence[str],
         max_concurrent_queries: int,
         total_concurrent_queries: int,
+        partition_count: int,
         stale_threshold_seconds: Optional[int],
         metrics: MetricsBackend,
         next_step: ProcessingStrategy[SubscriptionTaskResult],
         # TODO: To be removed once executor is fully rolled out
         # Commit is only passed here because we are temporarily
         # skipping executions during the transition phase.
         commit: Callable[[Mapping[Partition, Position]], None],
+        partitions: Mapping[Partition, int],
     ) -> None:
         self.__dataset = dataset
         self.__entity_names = set(entity_names)
-        self.__max_concurrent_queries = max_concurrent_queries
-        self.__total_concurrent_queries = total_concurrent_queries
+
+        use_calculated_value: bool = (
+            state.get_config("use_calculated_max_concurrent_queries", False) == True
+        )
+
+        calculated_max_concurrent_queries = self.__calculate_max_concurrent_queries(
+            partitions, partition_count, total_concurrent_queries, use_calculated_value
+        )
+        # XXX(meredith): temporarily log both the calculated and passed in
+        # max concurrent queries
+        metrics.gauge(
+            "calculated_max_concurrent_queries", calculated_max_concurrent_queries
+        )
+        metrics.gauge("max_concurrent_queries", max_concurrent_queries)
```

Comment:
Does this need to be sampled?

Reply:
I didn't think it did because this is only on initialization, should it be sampled?

Reply:
Ah so it is. No definitely not required then.